rfcs/rfcOVID-0018.md

```diff
@@ -0,0 +1,108 @@
+# No Longer Require a True Value at the End of a Module
+
+## Preamble
+
+    Author:  Curtis "Ovid" Poe <curtis.poe@gmail.com>
+    Sponsor:
+    ID:      0018
+    Status:  Draft
+
+## Abstract
+
+This RFC proposes a feature which, when used, causes the current Perl file to
+to yield a true value when C<require>d, eliminating the need for a "1" (or
+other true value) at the end of the file.
```

Comment:
No; this is a proposal that affects module compilation, not `require`. The primary usability distinction being where the feature needs to be used to be effective.

Reply:
That's not what's discussed in the 'Pre-RFC: yield true feature' p5p mail thread.

Reply:
That's an extremely long thread, and I can personally attest that it has been discussed there.

Reply:
Ah, it seems I haven't read Dave's proposal correctly. Switching between p5p and here is bad. 👎🏻 
I guess as it's an RfC now, continuing the discussing here is better.

Why would you remove the possibility to get the return value? Are there no use-cases for returning something from require? What about a module that is a singleton where loading returns it instead of having a constructor?

Reply:
Actually it's best to keep discussion about the design on p5p. And these questions have been discussed there (though as it's an extremely long thread, I understand missing them). The return value from require is largely useless because it only returns the value if the module has not been required before in that interpreter. Also, since this is an opt-in feature you can simply not use it. If you want to be able to use a return value from loaded code, you would need to use `do`.